x/ecocredit/server/testsuite/suite.go

```diff
@@ -112,40 +112,66 @@ func (s *IntegrationTestSuite) TestScenario() {
 
 	// Batch creation should fail if the StartDate is missing
 	err = (&ecocredit.MsgCreateBatchRequest{
-		Issuer:    issuer1,
-		ClassId:   clsID,
-		Issuance:  []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
-		StartDate: nil,
-		EndDate:   &time2,
+		Issuer:          issuer1,
+		ClassId:         clsID,
+		Issuance:        []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
+		StartDate:       nil,
+		EndDate:         &time2,
+		ProjectLocation: "AB",
 	}).ValidateBasic()
 	s.Require().Error(err)
 
 	// Batch creation should fail if the EndDate is missing
 	err = (&ecocredit.MsgCreateBatchRequest{
-		Issuer:    issuer1,
-		ClassId:   clsID,
-		Issuance:  []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
-		StartDate: &time1,
-		EndDate:   nil,
+		Issuer:          issuer1,
+		ClassId:         clsID,
+		Issuance:        []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
+		StartDate:       &time1,
+		EndDate:         nil,
+		ProjectLocation: "AB",
 	}).ValidateBasic()
 	s.Require().Error(err)
 
 	// Batch creation should fail if the EndDate is before the StartDate
 	err = (&ecocredit.MsgCreateBatchRequest{
-		Issuer:    issuer1,
-		ClassId:   clsID,
-		Issuance:  []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
-		StartDate: &time2,
-		EndDate:   &time1,
+		Issuer:          issuer1,
+		ClassId:         clsID,
+		Issuance:        []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
+		StartDate:       &time2,
+		EndDate:         &time1,
+		ProjectLocation: "AB",
+	}).ValidateBasic()
+	s.Require().Error(err)
+
+	// Batch creation should fail if the ProjectLocation is missing
+	err = (&ecocredit.MsgCreateBatchRequest{
+		Issuer:          issuer1,
+		ClassId:         clsID,
+		Issuance:        []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
+		StartDate:       &time1,
+		EndDate:         &time2,
+		ProjectLocation: "",
+	}).ValidateBasic()
+	s.Require().Error(err)
+
+	// Batch creation should fail if the ProjectLocation is invalid
+	err = (&ecocredit.MsgCreateBatchRequest{
+		Issuer:          issuer1,
+		ClassId:         clsID,
+		Issuance:        []*ecocredit.MsgCreateBatchRequest_BatchIssuance{},
+		StartDate:       &time1,
+		EndDate:         &time2,
+		ProjectLocation: "ABCD",
```

Comment:
just a side note, all those `ValidateBasic` tests should be moved to some unit tests for the `ecocredit` package as part of https://github.com/regen-network/regen-ledger/issues/423